runtime-components/daml-script-service/server/src/main/scala/com/daml/script/service/server/Server.scala

```diff
@@ -0,0 +1,169 @@
+  // TODO evaluate a sync and/or single-threaded approach for simplicity
+  private def newWorkStealingPool(): ExecutorService =
+    Executors.newWorkStealingPool(Runtime.getRuntime.availableProcessors())
+
+  private def newAkkaExecutionSequencerPool(implicit
+      system: ActorSystem
+  ): () => ExecutionSequencerFactory =
+    () =>
+      new AkkaExecutionSequencerPool(
+        poolName = ExecutionSequencerPoolName,
+        terminationTimeout = 5.seconds, // TODO make this configurable
+      )
+
+  // TODO handle exception-safe sanitazion and handling of `rawScriptId`
+  private def qualify(packageId: Ref.PackageId, rawScriptId: String): Identifier =
+    Identifier(packageId, QualifiedName.assertFromString(rawScriptId))
+
+  // TODO improve Daml Script itself to allow us to maintain a compiled script available
+  // TODO make this work with the Auth Middleware
+  private def connectAndRun(
+      input: JsValue,
+      dar: Dar[(Ref.PackageId, Ast.Package)],
+      rawScriptId: String,
+      port: Port,
+  )(implicit
+      ec: ExecutionContext,
+      esf: ExecutionSequencerFactory,
+      mat: Materializer,
+  ): Future[SValue] =
+    for {
+      clients <-
+        Runner.connect(
+          Participants[ApiParameters](
+            default_participant = Some(
+              ApiParameters(
+                host = InetAddress.getLoopbackAddress.getHostAddress, // TODO make this configurable
+                port = port.value, // TODO make this configurable
+                access_token = None, // TODO enable secure access, make this configurable
+                application_id = None, // TODO make this configurable
+              )
+            ),
+            participants = Map.empty, // TODO inquire: what is this?
+            party_participants = Map.empty, // TODO inquire: what is this?
+          ),
+          // TODO enable secure access, make `tlsConfig` configurable
+          tlsConfig = TlsConfiguration.Empty.copy(enabled = false),
+          maxInboundMessageSize = 4 * 1024 * 1024, // TODO make this configurable
+        )
+      value <- Runner.run(
+        dar = dar,
+        scriptId = qualify(dar.main._1, rawScriptId),
+        inputValue = Some(input),
+        initialClients = clients,
+        timeMode = ScriptTimeMode.WallClock, // TODO make this configurable
+      )
+    } yield value
+
+  // TODO open point: how to acquire and expose scripts?
+  private[server] val dar = DarDecoder.assertReadArchiveFromFile(darFile =
+    new File(BazelRunfiles.rlocation("daml-script/test/script-test.dar"))
+  )
+
+  private[server] def route(port: Port)(implicit
+      ec: ExecutionContext,
+      esf: ExecutionSequencerFactory,
+      mat: Materializer,
+  ): Route =
+    path(HttpRoutePrefix / Remaining) { scriptId =>
+      // TODO possibly index known scripts and explicitly route to a 404 for unknown scripts
+      // TODO evaluate a translation between URL and module separators, i.e. `/` -> `:`
```

Comment:
If you add an index page you could just use standard URL encoding for this.

Reply:
Good point, thanks.